src/silly_sprites/opengl/input.cpp

```diff
@@ -0,0 +1,114 @@
+        }
+    }
+    void Input::update_mouse_position(Window const& window) {
+        glfwGetCursorPos(window.get(), &mouse_position.first, &mouse_position.second);
+    }
+    [[nodiscard]] bool Input::is_quit() {
+        for (auto const& combination: close_input) {
+            auto close = true;
+            for (auto const& key : combination){
+                if (is_key_up(key)) { close = false; }
+            }
+            if (close) { return true; }
+        }
+        return false;
+    }
+
+    void Input::update(Window const& window) {
+        update_keys(window);
+        update_mouse(window);
+        update_mouse_position(window);
+        if (is_quit()) { glfwSetWindowShouldClose(window.get(), true); }
+    }
+
+    [[nodiscard]] Input::mouse_pos_ty Input::get_mouse_position() {
+        return mouse_position;
+    }
+
+    [[nodiscard]] bool Input::is_key_down(Keys key) {
+        return current_keys[static_cast<int>(key)];
+    }
+    [[nodiscard]] bool Input::is_key_up(Keys key) {
+        return not current_keys[static_cast<int>(key)];;
+    }
+
+    [[nodiscard]] bool Input::is_key_pressed(Keys key) {
+            return not last_keys[static_cast<int>(key)] and current_keys[static_cast<int>(key)];
+    }
+    [[nodiscard]] bool Input::is_key_released(Keys key) {
+            return last_keys[static_cast<int>(key)] and not current_keys[static_cast<int>(key)];
+    }
+
+    [[nodiscard]] bool Input::is_key_hold(Keys key) {
+            return last_keys[static_cast<int>(key)] and current_keys[static_cast<int>(key)];
+    }
+    [[nodiscard]] bool Input::is_key_raised(Keys key) {
+        return not last_keys[static_cast<int>(key)] and not current_keys[static_cast<int>(key)];
+    }
+
+    [[nodiscard]] bool Input::is_mouse_down(Mouse mouse) {
+        return current_mouse[static_cast<int>(mouse)];
+    }
+    [[nodiscard]] bool Input::is_mouse_up(Mouse mouse) {
+        return not current_mouse[static_cast<int>(mouse)];
+    }
+
+    [[nodiscard]] bool Input::is_mouse_pressed(Mouse mouse) {
+            return not last_mouse[static_cast<int>(mouse)] and current_mouse[static_cast<int>(mouse)];
+    }
+    [[nodiscard]] bool Input::is_mouse_released(Mouse mouse) {
+            return last_mouse[static_cast<int>(mouse)] and not current_mouse[static_cast<int>(mouse)];
+    }
+
+    [[nodiscard]] bool Input::is_mouse_hold(Mouse mouse) {
+            return last_mouse[static_cast<int>(mouse)] and current_mouse[static_cast<int>(mouse)];
+    }
+    [[nodiscard]] bool Input::is_mouse_raised(Mouse mouse) {
+        return not last_mouse[static_cast<int>(mouse)] and not current_mouse[static_cast<int>(mouse)];
+    }
+
+    [[nodiscard]] bool Input::add_close_input(std::vector<Keys> const& keys) {
+        
+        auto const contains{[keys](Keys key) {
+            for (auto const& k : keys) {
+                if (k == key) { return true; }
+            }
+            return false;
+        }};
```

Comment:
The styleguide required the use of `=`. Also, this lambda captures the vector by value (so it takes an expensive copy). It should capture the vector by reference. You can also utilize the `algorithm` header to simplify this.
```suggestion
        auto const contains = [&keys](Keys key) {
            return std::find(std::begin(keys), std::end(keys), key) != std::end(keys);
        };
```